src/components/UserSelector.tsx

```diff
@@ -1,34 +1,33 @@
-import React, { useContext, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
 import { User } from '../types/User';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import * as usersActions from '../features/users/users';
 
 type Props = {
   value: User | null;
   onChange: (user: User) => void;
 };
 
 export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
   value: selectedUser,
   onChange,
 }) => {
-  // `users` are loaded from the API, so for the performance reasons
-  // we load them once in the `UsersContext` when the `App` is opened
-  // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
   const [expanded, setExpanded] = useState(false);
 
+  const dispatch = useAppDispatch();
+  const { users } = useAppSelector(state => state.users);
+
+  useEffect(() => {
+    dispatch(usersActions.init());
+  }, [dispatch]);
+
   useEffect(() => {
     if (!expanded) {
       return;
     }
 
-    // we save a link to remove the listener later
     const handleDocumentClick = () => {
-      // we close the Dropdown on any click (inside or outside)
-      // So there is not need to check if we clicked inside the list
       setExpanded(false);
```

Comment:
Consider using `event.stopPropagation()` in the `handleDocumentClick` function to prevent the dropdown from closing when clicking inside the dropdown itself. This will improve the user experience by ensuring the dropdown only closes when clicking outside.